types/node/zlib.d.ts

```diff
@@ -160,6 +163,34 @@ declare module "zlib" {
     interface DeflateRaw extends stream.Transform, Zlib, ZlibReset, ZlibParams {}
     interface InflateRaw extends stream.Transform, Zlib, ZlibReset {}
     interface Unzip extends stream.Transform, Zlib {}
+    interface CompressDeflateInfoResult {
+        buffer: Buffer;
+        engine: Deflate;
+    }
+    interface CompressDeflateRawInfoResult {
+        buffer: Buffer;
+        engine: Deflate;
+    }
+    interface CompressGzipInfoResult {
+        buffer: Buffer;
+        engine: Gzip;
+    }
+    interface CompressGunzipInfoResult {
+        buffer: Buffer;
+        engine: Gunzip;
+    }
+    interface CompressInflateInfoResult {
+        buffer: Buffer;
+        engine: Inflate;
+    }
+    interface CompressInflateRawInfoResult {
+        buffer: Buffer;
+        engine: Inflate;
+    }
+    interface CompressUnzipInfoResult {
+        buffer: Buffer;
+        engine: Unzip;
+    }
```

Comment:
How about to create one type generic instead of 7 new interfaces?
```ts
type CompressInfoResult<Engine> = {
  buffer: Buffer;
  engine: Engine;
 }
 ```